esmvaltool/diag_scripts/hydrology/marrmot.py

```diff
@@ -0,0 +1,245 @@
+                                    units='J K-1 kg-1')
+
+    rd_const = iris.coords.AuxCoord(287.0,
+                                    long_name='Gas constant dry air',
+                                    units='J K-1 kg-1')
+
+    # Latent heat of vaporization in J kg-1 (or J m-2 day-1)
+    lambda_ = iris.coords.AuxCoord(2.5e6,
+                                   long_name='Latent heat of vaporization',
+                                   units='J kg-1')
+
+    # Specific heat of dry air constant pressure
+    cp_const = iris.coords.AuxCoord(1004,
+                                    long_name='Specific heat of dry air',
+                                    units='J K-1 kg-1')
+
+    beta = iris.coords.AuxCoord(20,
+                                long_name='Correction Constant',
+                                units='W m-2')
+
+    cs_const = iris.coords.AuxCoord(110,
+                                    long_name='Empirical constant',
+                                    units='W m-2')
+
+    # gamma = rv/rd * cp*msl/lambda_
+    # iris.exceptions.NotYetImplementedError: coord / coord
+    gamma = rv_const.points[0] / rd_const.points[0] * cp_const * psl / lambda_
+    return gamma, cs_const, beta, lambda_
+
+
+def debruin_pet(var_dict):
+    """ Determine De Bruin (2016) reference evaporation
+    Implement equation 6 from De Bruin (10.1175/JHM-D-15-0006.1)
+    """
+    # Unit checks:
+    psl = preproc.convert_units(var_dict['psl'], 'hPa')
+    tas = preproc.convert_units(var_dict['tas'], 'degC')
+
+    # Variable derivation
+    delta_svp = tetens_derivative(tas)
+    gamma, cs_const, beta, lambda_ = get_constants(psl)
+
+    # the definition of the radiation components according to the paper:
+    kdown = var_dict['rsds']
+    kdown_ext = var_dict['rsdt']
+    # Equation 6
+    rad_term = (1-0.23)*kdown - cs_const*kdown/kdown_ext
+    # the unit is W m-2
+    ref_evap = delta_svp / (delta_svp + gamma) * rad_term + beta
+
+    # 1 W m-2 = 86400 J m-2 day-1
+    pet = ref_evap * 86400 / lambda_
+    pet.var_name = 'potential_evapotranspiration'
+    pet.units = 'kg m-2 day-1'  # equivalent to mm/day
+    return pet
+
+
+def get_input_cubes(cfg):
+    """ Return a dict with all (preprocessed) input files """
+    provenance = create_provenance_record()
+    input_data = cfg['input_data'].values()
+    grouped_input_data = group_metadata(input_data,
+                                        'short_name',
+                                        sort='dataset')
+    all_vars = {}
+    for short_name in grouped_input_data:
+        logger.info("Loading variable %s", short_name)
+        input_files = [
+            attr['filename'] for attr in grouped_input_data[short_name]
+            ]
+        allyears = iris.load_cubes(input_files).concatenate_cube()
+        all_vars[short_name] = allyears
+        provenance['ancestors'].append(input_files)
+    return all_vars, provenance
+
+
+def _get_dataset_name(cfg):
```

Comment:
It would be better to re-write this function by a loop over all available datasets in the `main` function, see here for an example of how to do that:
https://github.com/ESMValGroup/ESMValTool/blob/baf2739eb92865c03d302dee02ea37566c0b13f8/esmvaltool/diag_scripts/hydrology/wflow.py#L205-L208

Reply:
Thanks, done.